src/main/kotlin/com/exactpro/th2/codec/openapi/writer/visitors/json/EncodeJsonObjectVisitor.kt

```diff
@@ -23,64 +23,94 @@ import com.exactpro.th2.codec.openapi.writer.SchemaWriter
 import com.exactpro.th2.codec.openapi.writer.visitors.SchemaVisitor.EncodeVisitor
 import com.exactpro.th2.common.grpc.Message
 import com.exactpro.th2.common.message.toJson
+import com.exactpro.th2.common.value.getBigDecimal
 import com.exactpro.th2.common.value.getDouble
 import com.exactpro.th2.common.value.getInt
 import com.exactpro.th2.common.value.getList
 import com.exactpro.th2.common.value.getLong
 import com.exactpro.th2.common.value.getMessage
 import com.exactpro.th2.common.value.getString
 import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.JsonNodeFactory
 import com.fasterxml.jackson.databind.node.ObjectNode
 import com.google.protobuf.ByteString
 import io.swagger.v3.oas.models.media.ArraySchema
 import io.swagger.v3.oas.models.media.Schema
+import java.math.BigDecimal
 
 
 class EncodeJsonObjectVisitor(override val from: Message) : EncodeVisitor<Message>() {
     private val rootNode: ObjectNode = mapper.createObjectNode()
 
-    override fun visit(fieldName: String, defaultValue: Schema<*>?, fldStruct: Schema<*>, required: Boolean) {
+    override fun visit(fieldName: String, defaultValue: Schema<*>?, fldStruct: Schema<*>, required: Boolean, schemaWriter: SchemaWriter) {
         from.getRequiredField(fieldName, required)?.getMessage()?.let { nextMessage ->
             val visitor = EncodeJsonObjectVisitor(nextMessage)
-            SchemaWriter.instance.traverse(visitor, fldStruct)
+            schemaWriter.traverse(visitor, fldStruct)
             rootNode.set<ObjectNode>(fieldName, visitor.rootNode)
         }
     }
 
     override fun visit(fieldName: String, defaultValue: String?, fldStruct: Schema<*>, required: Boolean) {
-        val value = from.getRequiredField(fieldName, required)?.getString()
-        fldStruct.checkEnum(value, fieldName)
-        rootNode.put(fieldName, value ?: defaultValue)
+        from.getRequiredField(fieldName, required)?.getString()?.let { value ->
+            fldStruct.checkEnum(value, fieldName)
+            rootNode.put(fieldName, value)
+        } ?: defaultValue?.let {
+            rootNode.put(fieldName, defaultValue)
+        }
     }
 
     override fun visit(fieldName: String, defaultValue: Boolean?, fldStruct: Schema<*>, required: Boolean) {
-        val value = from.getRequiredField(fieldName, required)?.getString()?.toBoolean()
-        fldStruct.checkEnum(value, fieldName)
-        rootNode.put(fieldName, value ?: defaultValue)
+        from.getRequiredField(fieldName, required)?.getString()?.toBoolean()?.let { value ->
+            fldStruct.checkEnum(value, fieldName)
+            rootNode.put(fieldName, value)
+        } ?: defaultValue?.let {
+            rootNode.put(fieldName, defaultValue)
+        }
```

Comment:
Can be extracted into a separate method. Also I'd suggest renaming `getRequiredField` method to `getField` since its current name implies that it always checks field presence